index.js

```diff
@@ -298,6 +304,77 @@ class GithubScm extends Scm {
         }
     }
 
+    /**
+     * Generate a deploy private and public key pair
+     * @async  generateDeployKey
+     * @return {Promise}                    Resolves to object containing the public and private key pair
+     */
+    async generateDeployKey() {
+        return new Promise((resolve, reject) => {
+            const location = DEPLOY_KEYS_FILE;
+            const comment = this.config.email;
+            const password = DEPLOY_KEYS_PASSWORD;
+            const format = DEPLOY_KEYS_FORMAT;
```

Comment:
could get rid of these vars and just define them inline in the object below, but not a big deal